src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -504,6 +505,17 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                         }
                         searchPattern = searchPattern.Substring(lastSlash + 1);
                     }
+
+                    // Typically we shouldn't see either of these cases
+                    if (searchPattern.IndexOfAny(s_EscapeCharacters) != -1)
```

Comment:
Whoops, missed your reply before I finished. I do think it is worth it as it wasn't super complicated and it saves some allocations.

I don't know that we can improve IndexOfAny without increasing overhead. Sounds like a good jitter optimization perhaps??